### PR TITLE
Bug 1908743: Add validation that the KubeMacPool component is enabled for the namespace

### PR DESCRIPTION
**Pull request by nirarg**

### CODE REVIEW

File: pkg/asset/installconfig/kubevirt/validation.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ func Validate(ic *types.InstallConfig, client Client) error {
 
 	allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(kubevirtPlatform.StorageClass, client, fldPath.Child("storageClass"))...)
 	allErrs = append(allErrs, validateNetworkAttachmentDefinitionExistsInNamespace(kubevirtPlatform.NetworkName, kubevirtPlatform.Namespace, client, fldPath.Child("networkName"))...)
+	allErrs = append(allErrs, validateKubeMacPoolComponentEnabledForNamespace(kubevirtPlatform.Namespace, client, fldPath.Child("KubeMacPoolComponent "))...)
```
Comment (bardielle) [inline pkg/asset/installconfig/kubevirt/validation.go:19]: There is a space in the string "KubeMacPoolComponent "
Are you sure it's working for you?

File: pkg/asset/installconfig/kubevirt/validation_test.go [hunk 2]
```diff
@@ -69,6 +71,8 @@ func TestKubevirtInstallConfigValidation(t *testing.T) {
 			expectClient: func(kubevirtClient *mock.MockClient) {
 				kubevirtClient.EXPECT().GetNetworkAttachmentDefinition(gomock.Any(), validNetworkName, validNamespace).Return(nil, nil).AnyTimes()
 				kubevirtClient.EXPECT().GetStorageClass(gomock.Any(), validStorageClass).Return(nil, nil).AnyTimes()
+				namespaceStruct.Labels = map[string]string{"mutatevirtualmachines.kubemacpool.io": "allocate"}
```
Comment (bardielle) [inline pkg/asset/installconfig/kubevirt/validation_test.go:74]: maybe you can move the map to be in generic part... it's the same map in all those test functions

File: pkg/asset/installconfig/kubevirt/validation.go (hunk 1 above)
Comment (nirarg) [reply pkg/asset/installconfig/kubevirt/validation.go:19]: Yes it worked (its only string), will remove

File: pkg/asset/installconfig/kubevirt/validation_test.go (hunk 2 above)
Comment (nirarg) [reply pkg/asset/installconfig/kubevirt/validation_test.go]: Defined the map as a variable

**Review by bardielle — APPROVED** [review]
/lgtm
/approve

File: pkg/asset/installconfig/kubevirt/validation.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ func Validate(ic *types.InstallConfig, client Client) error {
 
 	allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(kubevirtPlatform.StorageClass, client, fldPath.Child("storageClass"))...)
 	allErrs = append(allErrs, validateNetworkAttachmentDefinitionExistsInNamespace(kubevirtPlatform.NetworkName, kubevirtPlatform.Namespace, client, fldPath.Child("networkName"))...)
+	allErrs = append(allErrs, validateKubeMacPoolComponentEnabledForNamespace(kubevirtPlatform.Namespace, client, fldPath.Child("KubeMacPoolComponent"))...)
```
[the remark this answers is not in the document]
Comment (nirarg) [reply pkg/asset/installconfig/kubevirt/validation.go:19]: @staebler I'm aware of that, this is not validating a field provided by user, but it validate precondition user must follow
What do you suggest? can we proceed with that?
/hold

[the remark this answers is not in the document]
Comment (staebler) [reply pkg/asset/installconfig/kubevirt/validation.go:19]: The field that you are validating is the namespace.
```suggestion
	allErrs = append(allErrs, validateKubeMacPoolComponentEnabledForNamespace(kubevirtPlatform.Namespace, client, fldPath.Child("namespace"))...)
```

[the remark this answers is not in the document]
Comment (nirarg) [reply pkg/asset/installconfig/kubevirt/validation.go:19]: Its actually created to validate only the MacPoolComponent, I will change the validation to validate the namespace and than include this

[the remark this answers is not in the document]
Comment (nirarg) [reply pkg/asset/installconfig/kubevirt/validation.go]: Changed

**Pull request merged**